tests/test_circuit_breaker.py

```diff
@@ -0,0 +1,26 @@
+import unittest
+from unittest.mock import patch
+
+from pybreaker import CircuitBreakerError
+from requests.exceptions import HTTPError
+from canonicalwebteam.http import BaseSession
+
+
+def call_and_ignore_exceptions(fn, *args):
+    try:
+        fn(*args)
+    except HTTPError:
+        pass
+
+
+class TestCircuitBreaker(unittest.TestCase):
+    @patch("requests.Session.request")
+    def test_circuit_opens(self, request_mock):
+        request_mock.side_effect = HTTPError()
+        session = BaseSession()
+
+        for _ in range(4):
```

Comment:
``` python
In [6]: list(range(4))
Out[6]: [0, 1, 2, 3]
```

HTTP requests only fail 4 times before the `CircuitBreakerError`, which doesn't match the PR description:

> Adds a circuit-breaker to disable session requests **after** 5 consecutive failures.

I'm not sure whether the difference between 4 and 5 failures before the circuit breaker kicks in matters very much, but either way we should know what we're doing.

Reply:
When I looked at the snapcraft configuration and wrote the PR description I interpreted the `fail_max` parameter to be the allowed number of failures before the circuit opened, but when writing tests I realized it is actually opening at `fail_max - 1` failures. Do we want to update `fail_max` to 6, so we tolerate 5 failures? @nottrobin @tbille

Reply:
put it to 6 it should be fine.